piet-direct2d/src/lib.rs

```diff
@@ -418,6 +408,24 @@ impl<'a> RenderContext for D2DRenderContext<'a> {
 }
 
 impl<'a> D2DRenderContext<'a> {
+    fn fill_impl(&mut self, shape: impl Shape, brush: &impl IntoBrush<Self>, fill_rule: FillRule) {
+        let brush = brush.make_brush(self, || shape.bounding_box());
+
+        // TODO: do something special (or nothing at all) for line?
+        if let Some(rect) = shape.as_rect() {
```

Comment:
I wish there was an enum we could exhaustive match on so that anytime a new shape gets added you can't forget to handle it here. How likely do you think it is that new shape types will be added in the future?

Reply:
The theory is that failing to handle a shape is totally harmless, and not all back-ends will implement all shapes. Realistically I don't think the number will grow much.